mcs/class/corlib/System.Reflection.Emit/ILGenerator.cs

```diff
@@ -755,7 +755,7 @@ private void Emit (OpCode opcode, MethodInfo method, int token)
 			// Might be a DynamicMethod with no declaring type
 			Type declaringType = method.DeclaringType;
 			if (declaringType != null) {
-				if (declaringType.Module == module)
+				if (declaringType.Module == module || method is MethodBuilder)
```

Comment:
I guess the first case can happen without the second for already finished types.